mlflow/tracking/artifact_utils.py

```diff
@@ -72,3 +77,31 @@ def _download_artifact_from_uri(artifact_uri, output_path=None):
 
     return get_artifact_repository(artifact_uri=root_uri).download_artifacts(
         artifact_path=artifact_path, dst_path=output_path)
+
+
+def _upload_artifacts_to_databricks(source, run_id, source_host_uri=None,
+                                    target_databricks_profile_uri=None):
+    """
+    Copy the artifacts from ``source`` to the destination Databricks workspace (DBFS) given by
+    ``databricks_profile_uri`` or the current tracking URI.
+    :param source: Source location for the artifacts to copy.
+    :param run_id: Run ID to associate the artifacts with.
+    :param source_host_uri: Specifies the source artifact's host URI (e.g. Databricks tracking URI)
+        if applicable. If not given, defaults to the current tracking URI.
+    :param target_databricks_profile_uri: Specifies the destination Databricks host. If not given,
+        defaults to the current tracking URI.
+    :return: The DBFS location in the target Databricks workspace the model files have been
+        uploaded to.
+    """
+    import uuid
+    local_dir = tempfile.mkdtemp()
+    try:
+        source_with_profile = add_databricks_profile_info_to_artifact_uri(source, source_host_uri)
+        _download_artifact_from_uri(source_with_profile, local_dir)
+        dest_root = 'dbfs:/databricks/mlflow/tmp-external-source/'
```

Comment:
we might want to make it `'dbfs:/databricks/mlflow-tmp-external-source'` so it's not inside `dbfs:/databricks/mlflow` which is the root directory for default run artifact locations (non-ACL'd) default: 
+ Easier to distinguish as a separate kind of artifacts (to delete)
- Harder to discover (to delete) ?

Reply:
Yeah - because `create_model_version` is asynchronous (on Databricks due to the artifact copy), it's difficult/impossible for the client to do the deletion automatically. 

I'm thinking the current path is okay - we don't assume that all folders inside `dbfs:/databricks/mlflow/tmp-external-source` are run folders since it is completely user-writable.